pipelines/concourse.yml

```diff
@@ -1180,6 +1181,37 @@ jobs:
   - put: version
     params: {file: final-version/version}
 
+- name: discover-component-version
+  public: true
+  serial: true
+  plan:
+    - in_parallel:
+      - get: ci
+      - get: concourse
+        passed: [shipit]
+        trigger: true
+      - get: unit-image
+        passed: [shipit]
+        trigger: true
+      - get: version
+        passed: [shipit]
+    - task: discover-postgresql-version
+      file: ci/tasks/discover-component-version.yml
+      inputs: [version]
+      output_mapping: {component-version: postgresql-version}
+      params: {COMPONENT_NAME: "postgresql"}
+    - task: discover-helm-version
+      file: ci/tasks/discover-component-version.yml
+      inputs: [version]
+      output_mapping: {component-version: helm-version}
+      params: {COMPONENT_NAME: "helm"}
+    - put: resource-postgresql-version
+      inputs: [postgresql-version]
+      params: {file: "postgresql-version/postgresql-version-*.txt"}
+    - put: resource-helm-version
+      inputs: [helm-version]
+      params: {file: "helm-version/helm-version-*.txt"}
```

Comment:
It takes input as a version into the task, so the file name would have the Concourse version in it, so each Concourse version will have a unique file

Reply:
this might just be me being over-cautious, but i see in concourse/version and in release/5.7.x/version can both have the same version at some points, 

![Screen Shot 2020-04-21 at 8 35 16 AM](https://user-images.githubusercontent.com/2362482/79866630-126cfc00-83ab-11ea-82ef-a0c5807551b0.png)

![Screen Shot 2020-04-21 at 8 35 25 AM](https://user-images.githubusercontent.com/2362482/79866631-13059280-83ab-11ea-8449-4087583dbf80.png)

is that an issue?

if it's been fixed since 5.7, i apologize!

Reply:
(5.7.1-rc.7 through rc.9 are on both)

Reply:
ah, the ` discover-component-version` job has passed constraints on `shipit` and we wouldn't run `shipit` for the same version again across the pipelines.

Reply:
fair enough! thanks sammy!